File: lib/N.you_can_say_11.dart
void main() {
  print(isItAMultipleOfEleven("112233"));
  print(isItAMultipleOfEleven("30800"));
  print(isItAMultipleOfEleven("2937"));
  print(isItAMultipleOfEleven("323455693"));
  print(isItAMultipleOfEleven("5038297"));
  print(isItAMultipleOfEleven("112234"));
  print(isItAMultipleOfEleven("0"));
  print(isItAMultipleOfEleven(""));
}

/// todo 若一個數是11的倍數，那它的奇數項減偶數項也會是11的倍數。
bool isItAMultipleOfEleven(String num) {
  if (num.isEmpty) {
    return false;
  }

  var odd = 0;
  var even = 0;

  var isOdd = true;
  for (var i = 0; i < num.length; i++) {
    final value = int.parse(num[i]);

    if (isOdd) {
      odd += value;
    } else {
      even += value;
    }

    isOdd = !isOdd;
  }

  return 0 == odd - even;
}

File: src/N.you_can_say_11.kt
fun main() {
    println(isItAMultipleOfEleven("112233"))
    println(isItAMultipleOfEleven("30800"))
    println(isItAMultipleOfEleven("2937"))
    println(isItAMultipleOfEleven("323455693"))
    println(isItAMultipleOfEleven("5038297"))
    println(isItAMultipleOfEleven("112234"))
    println(isItAMultipleOfEleven("0"))
    println(isItAMultipleOfEleven(""))
}

fun isItAMultipleOfEleven(num: String): Boolean {
    if (num.isEmpty()) {
        return false
    }

    var odd = 0
    var even = 0

    var isOdd = true
    for (i in 0 ..< num.length) {
        val value = num[i].digitToInt()

        if (isOdd) {
            odd += value
        } else {
            even += value
        }

        isOdd = !isOdd
    }

    return 0 == odd - even
}
